quodlibet/quodlibet/ext/events/synchronizedlyrics.py

```diff
@@ -179,78 +215,184 @@ def _style_lyrics_window(self):
         """.format(self._get_background_color(), self._get_text_color(),
                    self._get_font_size()))
 
+    def _highlight_text(self):
+        return self._get_highlight_color()
+
     def _cur_position(self):
         return app.player.get_position()
 
     def _build_data(self):
+        cur = app.player.info
+        if cur is not None:
+            cur = SongWrapper(cur)
         self.text_buffer.set_text("")
         if app.player.song is not None:
             # check in same location as track
             track_name = app.player.song.get("~filename")
             new_lrc = os.path.splitext(track_name)[0] + ".lrc"
-            print_d("Checking for lyrics file %s" % new_lrc)
+            print_d("Checking for lyrics tag")
             if self._current_lrc != new_lrc:
                 self._lines = []
+                self._words = []
                 if os.path.exists(new_lrc):
-                    print_d("Found lyrics file: %s" % new_lrc)
-                    self._parse_lrc_file(new_lrc)
+                    print_d("Found lyrics file")
+                    self._parse_lrc_file(new_lrc, "file")
+                else:
+                    new_lrc = cur("~lyrics")
+                    print_d("Couldn't find lyrics file, defaulting to tags")
+                    self._parse_lrc_file(new_lrc, "tag")
             self._current_lrc = new_lrc
 
-    def _parse_lrc_file(self, filename):
-        with open(filename, 'r', encoding="utf-8") as f:
-            raw_file = f.read()
+    def _parse_lrc_file(self, lyrics, tagOrFile):
+        if tagOrFile == "file":
+            with open(lyrics, 'r', encoding="utf-8") as f:
+                raw_file = f.read()
+            self.raw_file_check = raw_file
+        elif tagOrFile == "tag":
+            raw_file = lyrics
+            raw_file.encode(encoding="utf-8")
```

Comment:
Shouldn't that be `raw_file = lyrics.encode(encoding="utf-8")` ? (I think currently the result is just thrown away.)